### PR TITLE
Send `watched` event

### PR DESCRIPTION
**Pull request by ironsidevsquincy**
Contains how much of the video was actually watched, in seconds

### CODE REVIEW

**Comment by JakeChampion** [comment]
@ironsidevsquincy Could you cover the changes in this PR with some tests?

**Review by commuterjoy — APPROVED** [review]

**Comment by commuterjoy** [comment]
Might fix some of the problems in #75 ?

**Comment by ironsidevsquincy** [comment]
@JakeChampion tests added. Had to update sinon to v2, as v1 doesn't actually support webpack (! -https://github.com/sinonjs/sinon#important-sinon-v1x-does-not-work-with-amdcommonjs-bundlers) so was failing when trying to use its timers

**Comment by ironsidevsquincy** [comment]
That progress issue is tough. won't fix it, but does add a (imo) more useful metric of time actually watched, rather than how far through the video people have gone (or seeked)

**Comment by commuterjoy** [comment]
I'd like to merge this please

**Comment by rowanbeentje** [comment]
I hoped there would be a better way to do this with ranges but nooo.  I see the caveat about seeking/replaying is already hinted at in the docs.

The only problem with this that I can see is re-use of the same player... looks like the playlist works fine, but won't this include ads in the played time on platforms where the same <video> element is used (eg iOS, Android)?

**Comment by rowanbeentje** [comment]
(I think it's worth blocking merge until ads issue/non-issue is confirmed, or there'll be misleading data)

**Comment by ironsidevsquincy** [comment]
@rowanbeentje, haven't seen it include ad time. Ads don't actually fire a `playing` event, which is the marker we use to decide if the video is being 'watched'

**Comment by ironsidevsquincy** [comment]
Playlist though is an interesting one. Will test...

**Comment by rowanbeentje** [comment]
Ah - thought playlist was fine, but yes, update() should also reset the time count (and probably fire the event or the data will be lost!)

Have you tested ad time on Android/iOS?  On those platforms video plays need a real click, so I think the Google ads library reuses the original <video> element for the ad as well, changing the src; looks to me like `playing` would fire in that situation...

**Comment by ironsidevsquincy** [comment]
@rowanbeentje, ok, now takes into account playlists (by resetting the time count)

Also, tested with a real click. Still doesn't fire a `playing` event if there is an ad (only when the actual video plays), so think we're good to go

**Review by rowanbeentje — APPROVED** [review]

**Review by JakeChampion — APPROVED** [review]

File: src/js/video.js [hunk 1]
```diff
@@ -23,15 +25,20 @@ function eventListener(video, ev) {
...
 	const event = new CustomEvent('oTracking.event', {
-		detail: {
-			action: ev.type,
-			advertising: video.opts.advertising,
+		detail: Object.assign({
 			category: 'video',
+			action,
+			advertising: video.opts.advertising,
 			contentId: video.opts.id,
-			progress: video.getProgress(),
```
Comment (JakeChampion) [inline src/js/video.js:32]: The fact that `progress` and `duration` are removed from the default event for o-video makes me think the release should be atleast a minor.

What do people think?

Comment (ironsidevsquincy) [reply src/js/video.js:32]: They haven't been removed from the existing events (all existing events behave the same), they're just not added to the new event this PR adds

**Pull request merged**